docs/how-to/use-rocm-agent-enumerator.rst

```diff
@@ -0,0 +1,281 @@
+              if device_id == target_device_id:
+                target_list.append(target)
+                break
+    
+      return target_list
+    
+    def readFromKFD():
+      target_list = []
+    
+      topology_dir = '/sys/class/kfd/kfd/topology/nodes/'
+      if os.path.isdir(topology_dir):
+        for node in sorted(os.listdir(topology_dir)):
+          node_path = os.path.join(topology_dir, node)
+          if os.path.isdir(node_path):
+            prop_path = node_path + '/properties'
+            if os.path.isfile(prop_path) and os.access(prop_path, os.R_OK):
+              target_search_term = re.compile("gfx_target_version.+")
+              with open(prop_path) as f:
+                try:
+                  line = f.readline()
+                except PermissionError:
+                  # We may have a subsystem (e.g. scheduler) limiting device visibility which
+                  # could cause a permission error.
+                  line = ''
+                while line != '' :
+                  search_result = target_search_term.search(line)
+                  if search_result is not None:
+                    device_id = int(search_result.group(0).split(' ')[1], 10)
+                    if device_id != 0:
+                      major_ver = int((device_id / 10000) % 100)
+                      minor_ver = int((device_id / 100) % 100)
+                      stepping_ver = int(device_id % 100)
+                      target_list.append("gfx" + format(major_ver, 'd') + format(minor_ver, 'x') + format(stepping_ver, 'x'))
+                  line = f.readline()
+    
+      return target_list
+    
+    def main():
+      if len(sys.argv) == 2 and sys.argv[1] == '-name' :
+        """ Prints the list of available AMD GCN target names extracted from rocminfo, a tool
+            shipped with this script to enumerate GPU agents available on a working ROCm stack."""
+        target_list = readFromROCMINFO(True) 
+      else:
+        """Prints the list of available AMD GCN ISA
+    
+        The program collects the list in 3 different ways, in the order of
+        precendence:
+    
+        1. ROCM_TARGET_LST : a user defined environment variable, set to the path and
+                           filename where to find the "target.lst" file. This can be
+                           used in an install environment with sandbox, where
+                           execution of "rocminfo" is not possible.
+        2. target.lst : user-supplied text file. This is used in a container setting
+                      where ROCm stack may usually not available.
+        3. HSA topology : gathers the information from the HSA node topology in
+                          /sys/class/kfd/kfd/topology/nodes/
+        4. lspci : enumerate PCI bus and locate supported devices from a hard-coded
+                   lookup table.
+        5. rocminfo : a tool shipped with this script to enumerate GPU agents
```

Comment:
This is confusing
is "rocminfo" different than ROCmInfo?

Reply:
Fixed.